matchms/similarity/CosineGreedy.py

```diff
@@ -83,14 +84,19 @@ def pair(self, reference: SpectrumType, query: SpectrumType) -> Tuple[float, int
         """
         def get_matching_pairs():
             """Get pairs of peaks that match within the given tolerance."""
-            matching_pairs = collect_peak_pairs(spec1, spec2, self.tolerance, shift=0.0,
-                                                mz_power=self.mz_power,
+            matching_pairs = collect_peak_pairs(spec1, spec2, self.tolerance,
+                                                shift=0.0, mz_power=self.mz_power,
                                                 intensity_power=self.intensity_power)
-            matching_pairs = sorted(matching_pairs, key=lambda x: x[2], reverse=True)
-            return matching_pairs
+            if matching_pairs is not None:
+                matching_pairs = numpy.array(matching_pairs)
+                matching_pairs = matching_pairs[numpy.argsort(matching_pairs[:, 2])[::-1], :]
+                return matching_pairs
+            return None
```

Comment:
Can you bailout early, thats easier to read for me
```suggestion
            if matching_pairs is None:
                return None
            matching_pairs = numpy.array(matching_pairs)
            matching_pairs = matching_pairs[numpy.argsort(matching_pairs[:, 2])[::-1], :]
            return matching_pairs
```

Reply:
Good point. Looks better that way.